controllers/machineregistration_controller_test.go

```diff
@@ -65,7 +77,7 @@ var _ = Describe("reconcile machine registration", func() {
 	})
 
 	AfterEach(func() {
-		Expect(test.CleanupAndWait(ctx, cl, mRegistration, setting)).To(Succeed())
+		Expect(test.CleanupAndWait(ctx, cl, mRegistration, setting, role, roleBinding, sa, secret)).To(Succeed())
```

Comment:
This should not be needed as far as I understand. The role, roleBinding, as and secret are all owned by the machineregistration, hence they should be deleted to by deleting the machine registration.

I am wondering what else are we missing here... 

What if we check objects are missing after `CleanupAndWait`, if there is an issue here then in should fail every time.

Reply:
Yeah, I figured out the child resources are not cleaned up in our unit-tests: I guess it is because of the test client we use, which doesn't take care of clean-up of owned resources.